spec/integration/post_spec.rb

```diff
@@ -0,0 +1,104 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :system do
+  user = User.create(name: 'John', post_counter: 30, photo: 'https://randomuser.me/api/portraits/men/70.jpg',
+                     bio: 'Teacher from Poland.')
+
+  subject do
+    Post.new(author_id: user.id, title: 'First Post', text: 'My first post', comments_counter: 20, likes_counter: 30)
+  end
+  before { subject.save }
+
+  describe 'index page' do
+    it "I can see the user's username." do
+      visit user_posts_path(user.id)
+      page.has_content?(user.name)
+    end
+
+    it "I can see the user's profile picture." do
+      visit user_posts_path(user.id)
+      page.has_css?('.img-fluid')
+    end
+
+    it 'I can see the number of posts the user has written.' do
+      visit user_posts_path(user.id)
+      page.has_content?(user.posts_counter)
```

Comment:
- [ ] Same here. Please use `post_counter` instead of `posts_counter`.

Reply:
Thanks @Omar-Muhamad for the corrections